sagemaker-fundamentals/create-training-job/container_kms/train.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+
+import boto3
+    
+import base64
+from cryptography.fernet import Fernet
+import os
+import json
+import pickle
+
+import pprint
+
+# init a boto3 session
+sess = boto3.Session(
+    aws_access_key_id=os.environ['ACCESS_KEY'],
+    aws_secret_access_key=os.environ['SECRET_KEY'],
+    region_name=os.environ['REGION_NAME']
+)
+
+# create s3 and kms client
+s3 = sess.client('s3')
+kms = sess.client('kms')
+
+
+# a dummy ML model
+class Model(object):
+    def __init__(self):
+        pass
+        
+
+# print out dictionary in a nice way
+pp = pprint.PrettyPrinter(indent=1)
+
+
+# where SageMaker injects training info inside container
+input_dir="/opt/ml/input/"
+
+# where SageMaker injects train / validation /test data
+data_dir = '/opt/ml/input/data/'
+
+# SageMaker treat "/opt/ml/model" as checkpoint direcotry
+# and it will send everything there to S3 output path you 
+# specified 
+model_dir="/opt/ml/model"
+
+
+def encrypt(data:bytes, plaintext_key:bytes) -> bytes:
+    """Encrypt a chunk of bytes on client-side
+    
+    Args:
+        data: a chunk of bytes
+        plaintext_key: plaintext data key
+        
+    Return:
+        encrypted data
+    """
+    ascii_str = base64.b64encode(plaintext_key)
+    f = Fernet(key=ascii_str) 
+    return f.encrypt(data)
+
+
+def decrypt(data:bytes, plaintext_key:bytes) -> bytes:
+    """Decrypt a chunk of bytes on client-side
+    
+    Args:
+        data: encypted binary data
+        plaintext_key: plaintext data key
+    
+    Return:
+        decrypted data
+    """    
+    # to Fernet-friendly key
+    ascii_str = base64.b64encode(plaintext_key)
+    
+    f = Fernet(key=ascii_str)
+    return f.decrypt(data)
```

Comment:
Just use the AWS Encryption SDK
https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/python-example-code.html#python-example-streams